Added new TI devices

eto- opened the pull request:
##### SUMMARY
<!-- Describe the changes you have made. -->
Added new TI devices with dependencies.

##### DATASHEETS / REFERENCES
<!-- When submitting a new device or package, please provide links to the
necessary datasheets, package drawings or standards documents.-->
All the links are in the datasheet field of the corresponding devices

##### OPEN QUESTIONS / UNRESOLVED ISSUES
<!-- List topics you are unsure about, if any. Otherwise, leave this empty. -->
- The quality of the step files may be improved.
- The legend for high density packages is complicated
- The WSON are nonstandard packages from TI, therefore I added them here.
- The WQFN10 is more standard and it could go in Base library but it would be better to have a script generating them
Symbols and Components can be moved elsewhere if needed.


##### CHECKLIST
<!-- Please review the following checklist and check the items you agree with
by adding an "x" between the brackets (i.e. "[x]"). If you are unsure, leave
the corresponding items unchecked so the library maintainers know where help
is needed. -->

- [x ] I have read and followed the library conventions¹.
- [ ] For packages, I followed IPC7351C (see details in library conventions).
- [ ] For modified elements, I bumped their version number².
- [x ] I already used the new library elements in a design and verified their correctness (optional).
- [x ] I'm the copyright owner of the added content (i.e. the changes are made by myself, not copied/imported from somewhere else).
- [x ] I agree to publish all my changes under the CC0 Public Domain License³, allowing everyone to use and modify the content without any restrictions.

¹ *Library Conventions: https://docs.librepcb.org/#libraryconventions*
² *Minor version bump if only metadata was modified (e.g. "0.1" -> "0.1.1"), major version bump if functional changes were made (e.g. "0.1" -> "0.2")*
³ *CC0 Public Domain License: https://en.wikipedia.org/wiki/CC0*

eto- commented:
Updates?

ubruhin commented:
Huh that's a lot to review :see_no_evil: 

Would you mind splitting this into separate PRs? For the start, a PR adding only the devices which don't need new dependencies would be nice - they are straight-forward to review so we can get them merged fast. Then we can do the rest step by step.

Btw, thanks for using the new feature of adding datasheet URLs, it is very useful for the review process :smiley:

eto- commented:
How do you prefer me to proceed?

I can unwind my commit and split the PRs between:
- the simple devices that I added
- the wson devices
- the wqfn10 devices

Let me know

ubruhin commented:
> I can unwind my commit and split the PRs between:

Yes that's what I meant in my comment above. You could also just move the simple devices to a separate branch and open a PR for it and we pick up the rest later. Up to you.

For the new packages we'll have to decide anyway how to proceed with them (base library or not, generator or not).

eto- commented:
I created the branch for simple devices and started a new PR.

The packages are 2 types: 
  - wson are TI specific, I do not think it is worth to have generator
  - WQFN are generic and there is a generator script.
